### PR TITLE
Added support for a new version of the TF OD API pre-processing part

### PR DESCRIPTION
**Pull request by lazarevevgeny**
Description: Added support for a new version of the TF OD API pre-processing part. Refer to the ticket for more details.

JIRA: 37922

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR
* [x]  Transformation preserves original framework node names: N/A for this type of transformation

Validation:
* [x]  Unit tests: N/A nothing to cover
* [x]  Framework operation tests: N/A no new operation enabled
* [x]  Transformation tests: N/A for this type of transformation
* [x]  e2e model test with an update of ./tests/e2e_oss/utils/reshape_utils.py: N/A no new public model enabled
* [x]  Model Optimizer IR Reader check: Done manually

Documentation:
* [x]  Supported frameworks operations list: N/A no new op enabled
* [x]  Supported **public** models list: N/A no new model enabled
* [x]  New operations specification: N/A no new op enabled
* [x]  Guide on how to convert the **public** model: N/A no changes needed
* [x]  User guide update: N/A no changes needed

### CODE REVIEW

File: model-optimizer/extensions/front/tf/ObjectDetectionAPI.py [hunk 1]
```diff
@@ -486,6 +486,22 @@ def nodes_to_remove(self, graph: Graph, match: SubgraphMatch):
...
+
+        :param to_float: the Cast node which converts the input tensor to Float
+        :param mul: the Mul node (can be None)
+        :param sub: the Sub node
+        :return: the result of the check
+        """
+        if mul is not None:
+            return any([port.node.id == mul.id for port in to_float.out_port(0).get_destinations()])
+        else:
+            return any([port.node.id == sub.id for port in to_float.out_port(0).get_destinations()])
```
Comment (rkazants) [inline model-optimizer/extensions/front/tf/ObjectDetectionAPI.py:503]: I think that assert for `sub`  that it is not None is needed

File: model-optimizer/extensions/front/tf/ObjectDetectionAPI.py [hunk 2]
```diff
@@ -486,6 +486,22 @@ def nodes_to_remove(self, graph: Graph, match: SubgraphMatch):
                 new_nodes_to_remove.remove(node_to_keep)
         return new_nodes_to_remove
 
+    def is_preprocessing_applied_before_resize(self, to_float: Node, mul: Node, sub: Node):
+        """
+        The function checks if the output of 'to_float' operation which is consumed by 'mul' or 'sub'. If this is true
```
[the remark this answers is not in the document]
Comment (lazarevevgeny) [reply model-optimizer/extensions/front/tf/ObjectDetectionAPI.py]: Fixed.

File: model-optimizer/extensions/front/tf/ObjectDetectionAPI.py (hunk 1 above)
Comment (lazarevevgeny) [reply model-optimizer/extensions/front/tf/ObjectDetectionAPI.py:504]: Fixed.

File: model-optimizer/extensions/front/tf/ObjectDetectionAPI.py [hunk 3]
```diff
@@ -486,6 +486,23 @@ def nodes_to_remove(self, graph: Graph, match: SubgraphMatch):
                 new_nodes_to_remove.remove(node_to_keep)
         return new_nodes_to_remove
 
+    def is_preprocessing_applied_before_resize(self, to_float: Node, mul: Node, sub: Node):
+        """
+        The function checks if the output of 'to_float' operation is consumed by 'mul' or 'sub'. If this is true that
+        means that the pre-processing (mean/scale) is applied before the image resize. The image resize was applied
```
Comment (rkazants) [inline model-optimizer/extensions/front/tf/ObjectDetectionAPI.py:492]: If this is true that means that the pre-processing (mean/scale) is applied before the image resize. ->
If this is true, the pre-processing (mean/scale) is applied before the image resize.

**Review by rkazants — COMMENTED** [review]
Does ObjectDetectionAPI_test.py need update since new case (resize applied after) has appeared?

**Comment by lazarevevgeny** [comment]
> Does ObjectDetectionAPI_test.py need update since new case (resize applied after) has appeared?

We don't have any test for TF OD API models rather than e2e test due to complexity of the code and many dependencies between the transformations used. Right now we don't have public models to add to e2e so the e2e test cannot be created also. So I just checked the solution using two models which I have.

File: model-optimizer/extensions/front/tf/ObjectDetectionAPI.py (hunk 3 above)
Comment (lazarevevgeny) [reply model-optimizer/extensions/front/tf/ObjectDetectionAPI.py]: Fixed

**Review by rkazants — APPROVED** [review]

**Review by mvafin — APPROVED** [review]

**Pull request merged**